### PR TITLE
iOS: Bullet proof header search paths

### PR DESCRIPTION
**Pull request by romainbriche**
Support both the use or non use of `use_frameworks!` in the target `Podfile`.

Prevents all those `#import <> not found` issues.

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
 	"private": true,
 	"scripts": {
 		"start": "node node_modules/react-native/local-cli/cli.js start",
-		"test": "jest"
+		"test": "jest",
+		"postinstall": "rm -rf node_modules/react-native-analytics-segment-io/example"
```
Comment (kevinpelgrims) [inline example/package.json]: Don't forget to update the documentation if Yarn is now the preferred way to work with the example app (https://github.com/leoilab/react-native-analytics-segment-io#example).

**Comment by kevinpelgrims** [comment]
Just to be very explicit about this here: There is no change in how the library is consumed, correct?

**Comment by romainbriche** [comment]
No. It works exactly as it used to be

File: example/package.json (hunk 1 above)
Comment (romainbriche) [reply example/package.json]: Good point 👍

**Pull request merged**